python/tk_desktop/rpc.py

```diff
@@ -44,12 +44,40 @@
 # type that is stored in the pickle and assumes the output of a dump
 # can always be turned into a utf-8 encoded string, which is false.
 # Therefore, we'll simply take the raw pickle and send it over the wire.
-from sgtk.util import pickle
-from tank_vendor.six.moves.BaseHTTPServer import HTTPServer
-from tank_vendor.six.moves.SimpleHTTPServer import SimpleHTTPRequestHandler
-from tank_vendor.six.moves.urllib.request import urlopen
 from tank_vendor import six
 from tank_vendor.six.moves import cPickle as py_pickle
+from tank.util import pickle as tk_pickle, is_windows
```

Comment:
This pretty much reverts the file to the state it was before the Python 3 port. What I've done in this branch is basically restore the original file and report it to Python 3.

Reply:
You're quite right actually. I did revert to the original file. When I typed this I had forgotten that I had reversed course and restored to the ported code and stripped things out instead. This is because while writing the tests I had actually found some bugs in the RPC code, so I kept those fixes in this final version and simply tweaked about the recv/send operations were done.